src/main/java/tc/oc/pgm/match/MatchManagerImpl.java

```diff
@@ -31,15 +37,32 @@
 import tc.oc.util.logging.ClassLogger;
 import tc.oc.world.NMSHacks;
 
+@SuppressWarnings("UnstableApiUsage")
 public class MatchManagerImpl implements MatchManager, MultiAudience {
 
   private final Logger logger;
   private final Server server;
   private final MapLibrary library;
   private final MapLoader loader;
 
-  private final Map<String, Match> matchById;
-  private final Map<String, String> matchIdByWorldName;
+  private final Map<String, Match> matchById = new HashMap<>();
+  private final Map<String, String> matchIdByWorldName = new HashMap<>();
+  private final LoadingCache<PGMMap, String> preMatch =
+      CacheBuilder.newBuilder()
+          .expireAfterWrite(10, TimeUnit.SECONDS)
+          // If after 10 secs of writing the match it hasn't been loaded, unload & destroy it
+          .removalListener(
+              (RemovalNotification<PGMMap, String> r) -> {
+                if (!matchById.get(r.getValue()).isLoaded()) unloadMatch(r.getValue());
```

Comment:
Don't you need to check `r.isEvicked()` ?

Reply:
I tried to find such a method, but all i had found is `r.getCause()` wich has a non-public `wasEvicted`. Turns out RemovalNotification exposes this via `r.wasEvicted()`.
I think however, checking if the map was loaded at all, is safer in regards to not leaving non-used matches floating arround and never unable to fully get removed.